File: app/src/main/java/space/levan/memory/di/component/AppComponent.java
package space.levan.memory.di.component;

import javax.inject.Singleton;

import dagger.Component;
import space.levan.memory.app.App;
import space.levan.memory.di.module.AppModule;
import space.levan.memory.di.module.HttpModule;
import space.levan.memory.model.DataManager;
import space.levan.memory.model.db.DBHelperImpl;
import space.levan.memory.model.http.HttpHelperImpl;
import space.levan.memory.model.prefs.SPrefsHelperImpl;

/**
 * Application component
 *
 * @author WangZhiYao
 * @date 2017/10/17
 */

@Singleton
@Component(modules = {AppModule.class, HttpModule.class})
public interface AppComponent {

    /**
     * get Context
     *
     * @return
     */
    App getContext();

    /**
     * get DataManager
     *
     * @return
     */
    DataManager getDataManager();

    /**
     * get DBHelper
     *
     * @return
     */
    DBHelperImpl getDBHelper();

    /**
     * get HttpHelper
     *
     * @return
     */
    HttpHelperImpl getHttpHelper();

    /**
     * get SPrefsHelper
     *
     * @return
     */
    SPrefsHelperImpl getSPrefsHelper();
}

File: app/src/main/java/space/levan/memory/di/component/AppComponent.kt
package space.levan.memory.di.component

import dagger.Component
import space.levan.memory.App
import space.levan.memory.di.module.AppModule
import space.levan.memory.di.module.HttpModule
import javax.inject.Singleton

/**
 * File description
 *
 * @author WangZhiYao
 * @date 2018/1/15
 */

@Singleton
@Component(modules = arrayOf(AppModule::class, HttpModule::class))
interface AppComponent {

    var context: App
}